extensions/modules/src/org/exist/xquery/modules/compression/DeflateFunction.java

```diff
@@ -0,0 +1,113 @@
+/*
+ *  eXist Open Source Native XML Database
+ *  Copyright (C) 2018 The eXist Project
+ *  http://exist-db.org
+ *
+ *  This program is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public License
+ *  as published by the Free Software Foundation; either version 2
+ *  of the License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * $Id$
+ */
+package org.exist.xquery.modules.compression;
+
+import java.io.ByteArrayInputStream;
```

Comment:
Could you use the eXist-db variant `org.exist.util.io.FastByteArrayInputStream` please?

Reply:
ow yea, sorry I forgot about that one..... does it have an efficient `toInputStream()` too? (meaning: no serialization to another `byte[]`)

Reply:
@dizzzz `FastByteArrayInputStream` is an `InputStream`, so why would you call `toInputStream`?

Reply:
ahhhh sorry, coffee time.....